### PR TITLE
Enhancements/travis ci deployments

### PR DESCRIPTION
**Pull request by Kinlaw**
Pushes build files to build repo.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,6 +8,19 @@ notifications:
...
+- openssl aes-256-cbc -K $encrypted_8e579b44016d_key -iv $encrypted_8e579b44016d_iv -in deploy_key.enc -out deploy_key -d
+
+- chmod 600 deploy_key
+- eval `ssh-agent -s`
+- ssh-add deploy_key
+after_success:
+- .travis/after_success.sh
+env:
+  global:
+  - REPO="git@github.com:RedHatInsights/insights-deployment-test-build"
```
Comment (ryelo) [inline .travis.yml]: Are we updating this to `insights-dashboard-build`?

File: .travis.yml [hunk 2]
```diff
@@ -8,6 +8,19 @@ notifications:
...
+  - FILES="dist/index.html"
+  - GH_REPO="https://github.com/RedHatInsights/insights-deployment-test"
```
Comment (ryelo) [inline .travis.yml]: Same as above, but `insights-dashboard`

File: .travis/after_success.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    echo -e "Pull Request, not pushing a build"
+    exit 0;
+fi
+
+# If current dev branch is master, push to build repo master
+if [ "${TRAVIS_BRANCH}" = "master" ]; then
+    .travis/release.sh
+fi
+
+# If current branch dev branch is stable/foo, push to build repo stable
```
Comment (ryelo) [inline .travis/after_success.sh]: @Kinlaw can you update this too in your next commit? Should say "# If current dev branch is stable/foo, push to build repo stable`

Remove the first occurrence of "branch"

File: .travis.yml (hunk 1 above)
Comment (Kinlaw) [reply .travis.yml]: uhwhoops

Comment (Kinlaw) [reply .travis.yml]: actually I don't think these variables are used any longer. I'm removing them.

Comment (ryelo) [reply .travis.yml]: They're referenced in the `.sh` files, so don't remove :)

File: .travis/release.sh [hunk 4]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+MESSAGE=$(git log --format=%B -n 1 $TRAVIS_COMMIT)
+cd dist
+git init
+git config --global user.name $COMMIT_AUTHOR_USERNAME
+git config --global user.email $COMMIT_AUTHOR_EMAIL
+git remote add travis-build ${REPO}.git
+git add .
+git commit -m 'Build by Travis'
```
Comment (ryelo) [inline .travis/release.sh]: Should we reference the `$COMMIT_AUTHOR_USERNAME` in the commit message, too?

**Comment by Kinlaw** [comment]
Changes implemented

File: .travis.yml [hunk 5]
```diff
@@ -8,6 +8,17 @@ notifications:
...
 - '10'
 install:
-  - npm ci
+- npm ci
 script:
-  - npm run travis:verify
+- npm run travis:verify
+before_install:
+# adds deploy_key for pushing to ${REPO}
+- openssl aes-256-cbc -K $encrypted_8e579b44016d_key -iv $encrypted_8e579b44016d_iv -in deploy_key.enc -out ~\/.ssh/deploy_key -d
```
Comment (ryelo) [inline .travis.yml]: Travis is failing on this command right now

Comment (ryelo) [inline .travis.yml]: Travis is failing on this command

**Pull request merged**